drivers/tests/integration_tests.rs

```diff
@@ -29,9 +38,271 @@ fn run_driver_test(test_bin_name: &'static str) {
+
+        fn swap_word_bytes(words: &[u32]) -> Vec<u32> {
+            words.iter().map(|word| word.swap_bytes()).collect()
+        }
+        fn swap_word_bytes_inplace(words: &mut [u32]) {
+            for word in words.iter_mut() {
+                *word = word.swap_bytes()
+            }
+        }
+        fn hmac384(key: &[u8], data: &[u8]) -> [u8; 48] {
+            let pkey = PKey::hmac(key).unwrap();
+            let mut signer = Signer::new(MessageDigest::sha384(), &pkey).unwrap();
+            signer.update(data).unwrap();
+            let mut result = [0u8; 48];
+            signer.sign(&mut result).unwrap();
+            result
+        }
+        fn aes256_decrypt_blocks(key: &[u8], iv: &[u8], input: &[u8]) -> Vec<u8> {
+            let cipher = Cipher::aes_256_cbc();
+            let mut ctx = CipherCtx::new().unwrap();
+            ctx.decrypt_init(Some(cipher), Some(key), Some(iv)).unwrap();
+            ctx.set_padding(false);
+            let mut result = vec![];
+            ctx.cipher_update_vec(input, &mut result).unwrap();
+            ctx.cipher_final_vec(&mut result).unwrap();
+            result
+        }
+
+        let mut result = Self::default();
+
+        result
+            .uds
+            .as_bytes_mut()
+            .copy_from_slice(&aes256_decrypt_blocks(
+                swap_word_bytes(&input.doe_obf_key).as_bytes(),
+                &input.doe_iv,
+                swap_word_bytes(&input.uds_seed).as_bytes(),
+            ));
+        swap_word_bytes_inplace(&mut result.uds);
+
+        result.expected_test_results.hmac_uds_as_key = transmute!(hmac384(
+            swap_word_bytes(&result.uds).as_bytes(),
+            "Hello world!".as_bytes()
+        ));
+        swap_word_bytes_inplace(&mut result.expected_test_results.hmac_uds_as_key);
+
+        result.expected_test_results.hmac_uds_as_data = transmute!(hmac384(
+            swap_word_bytes(&caliptra_drivers_test_bin::DOE_TEST_HMAC_KEY).as_bytes(),
+            swap_word_bytes(&result.uds).as_bytes()
+        ));
+        swap_word_bytes_inplace(&mut result.expected_test_results.hmac_uds_as_data);
+
+        // After reset, the key-vault registers are filled with a particular
+        // word, depending on the debug-locked mode.  The field entropy only
+        // takes up 8 of the 12 words, so the 4 remaining words keep their
+        // original value (their contents are used when the key-vault entry is
+        // used as a HMAC key, but not when used as HMAC
+        // data).
+        result.field_entropy = [input.keyvault_initial_word_value; 12];
+        result.field_entropy[0..8]
+            .as_bytes_mut()
+            .copy_from_slice(&aes256_decrypt_blocks(
+                swap_word_bytes(&input.doe_obf_key).as_bytes(),
+                &input.doe_iv,
+                swap_word_bytes(&input.field_entropy_seed).as_bytes(),
+            ));
+        swap_word_bytes_inplace(&mut result.field_entropy);
```

Comment:
Might be a bit cleaner if the word swapping was taken care of in `aes256_decrypt_blocks()` and `hmac384()`.

Reply:
I was trying to make it clear to the reader where things differ from the standard byte order, which those aes256_decrypt_blocks() and hmac384() use.

Reply:
Also, doe_iv doesn't have its words swapped (to be consistent with the doe driver), and neither does the buffer data passed to hmac.